src/config-import.js

```diff
@@ -0,0 +1,271 @@
+          addTags: {},
+          removeTags: {},
+          terms: [],
+        }
+        for (const key of Object.keys(valueSchemas.preset.properties)) {
+          if (key in preset) {
+            // @ts-ignore - we validate below
+            presetValue[key] = preset[key]
+          }
+        }
+        if (!validate('preset', presetValue)) {
+          warnings.push(new Error(`Invalid preset ${preset.name}`))
+          continue
+        }
+        yield {
+          fieldNames:
+            'fields' in preset && Array.isArray(preset.fields)
+              ? preset.fields
+              : [],
+          iconName:
+            'icon' in preset && typeof preset.icon === 'string'
+              ? preset.icon
+              : undefined,
+          value: presetValue,
+        }
+      }
+    },
+  }
+}
+
+/**
+ * @param {string} filename
+ * @param {Buffer} buf
+ * @returns {{ name: string, variant: IconData['variants'][Number] }}}
+ */
+function parseIcon(filename, buf) {
+  const parsedFilename = path.parse(filename)
+  const matches = parsedFilename.base.match(
+    /([a-zA-Z0-9-]+)-([a-zA-Z]+)@(\d+)x\.(png|svg)$/
+  )
+  if (!matches) {
+    throw new Error(`Unexpected icon filename ${filename}`)
+  }
+  /* eslint-disable no-unused-vars */
+  const [_, name, size, pixelDensityStr, extension] = matches
+  const pixelDensity = Number(pixelDensityStr)
+  if (!(pixelDensity === 1 || pixelDensity === 2 || pixelDensity === 3)) {
+    throw new Error(`Error loading icon. invalid pixel density ${pixelDensity}`)
+  }
+  if (!(size === 'small' || size === 'medium' || size === 'large')) {
+    throw new Error(`Error loading icon. invalid size ${size}`)
+  }
+  if (!name) {
+    throw new Error('Error loading icon. missing name')
+  }
+  /** @type {'image/png' | 'image/svg+xml'} */
+  let mimeType
+  switch (parsedFilename.ext.toLowerCase()) {
+    case '.png':
+      mimeType = 'image/png'
+      break
+    case '.svg':
+      mimeType = 'image/svg+xml'
+      break
+    default:
+      throw new Error(`Unexpected icon extension ${parsedFilename.ext}`)
+  }
+  return {
+    name,
+    variant: {
+      size,
+      mimeType,
+      pixelDensity,
+      blob: buf,
+    },
+  }
+}
+
+/**
+ * @param {unknown} presetsFile
+ * @returns {asserts presetsFile is PresetsFile}
+ */
+function validatePresetsFile(presetsFile) {
+  if (
+    !isRecord(presetsFile) ||
+    !('presets' in presetsFile) ||
+    typeof presetsFile.presets !== 'object' ||
+    !('fields' in presetsFile) ||
+    typeof presetsFile.fields !== 'object'
+  ) {
+    throw new Error('Invalid presets.json file')
+  }
+}
+
+/**
+ * @param {unknown} value
+ * @returns {value is Record<string, unknown>}
+ */
+function isRecord(value) {
+  return value !== null && typeof value === 'object' && !Array.isArray(value)
+}
+
+/**
+ * @param {Record<string | symbol, unknown>} obj
+ * @param {string | symbol} prop
+ */
+function hasOwn(obj, prop) {
+  if (!isRecord(obj)) return
```

Comment:
Nit: why is this check necessary?

```suggestion
```

Reply:
I need to double check, but if obj is null, then `hasOwnPropery.call` will fail (and we call hasOwn with stuff that can be null)